packages/examples/cvat/exchange-oracle/src/services/cvat.py

```diff
@@ -515,17 +515,24 @@ def get_free_job(
     session: Session,
     cvat_projects: list[int],
     *,
+    user_wallet_address: str,
     for_update: bool | ForUpdateParams = False,
 ) -> Job | None:
+    """
+    Returns the first available job that wasn't previously assigned to that user_walled_address.
+    """
     return (
         _maybe_for_update(session.query(Job), enable=for_update)
         .where(
             Job.cvat_project_id.in_(cvat_projects),
             Job.status == JobStatuses.new,
             ~Job.assignments.any(
-                (Assignment.status == AssignmentStatuses.created.value)
-                & (Assignment.completed_at == None)
-                & (utcnow() < Assignment.expires_at)
+                (
+                    (Assignment.status == AssignmentStatuses.created.value)
+                    & (Assignment.completed_at == None)
+                    & (utcnow() < Assignment.expires_at)
+                )
+                | (Assignment.user_wallet_address == user_wallet_address),
```

Comment:
What if it is assigned to user1, after that to user2, and again to user1. Will user1 get it second time?

Reply:
This part of expression roughly translates to: look for job that has no previous assignments with this wallet id. 

So, answering your question, no, once job has been assigned to a user, it will never be assigned to that user again.